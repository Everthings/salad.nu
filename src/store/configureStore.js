import { configureStore, getDefaultMiddleware } from "@reduxjs/toolkit";
import rootReducer from "./reducer";
import api from "./middleware/api";

export default function () {
  return configureStore({
    reducer: rootReducer,
    middleware: [...getDefaultMiddleware(), api],
  });
}
