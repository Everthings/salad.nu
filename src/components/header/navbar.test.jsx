import React from "react";
import { Provider } from "react-redux";
import { BrowserRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";
import { cleanup } from "@testing-library/react";
import configureStore from "redux-mock-store";
import Theme from "../themes/theme";
import NavBar from "./navbar";
import salad_logo from "./../../images/cover_no_background.png";

const initialState = {
  entities: {
    interactions: { searchStr: "" },
    term: { name: "2019 Fall" },
    schedule: { list: [] },
    theme: { currentTheme: "green" },
  },
};
const mockStore = configureStore([]);

afterEach(cleanup);

describe("NavBar", () => {
  it("renders correctly with logo", () => {
    const store = mockStore(initialState);
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router>
            <Theme>
              <NavBar logo={salad_logo} />
            </Theme>
          </Router>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly without logo", () => {
    const store = mockStore(initialState);
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router>
            <Theme>
              <NavBar />
            </Theme>
          </Router>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
