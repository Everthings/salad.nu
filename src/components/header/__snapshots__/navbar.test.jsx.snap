// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar renders correctly with logo 1`] = `
<nav
  className="sc-AxhCb dYJNSj navbar navbar-light"
>
  <div
    className="flex-fill"
  >
    <form
      className="mr-3"
    >
      <input
        aria-label="Search"
        autoFocus={true}
        className="sc-AxjAm gGRDat form-control"
        data-testid="search-bar"
        onChange={[Function]}
        placeholder="Search Courses "
        type="search"
        value=""
      />
    </form>
  </div>
  <div
    className="react-toggle react-toggle--checked"
    onClick={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
  >
    <div
      className="react-toggle-track"
    >
      <div
        className="react-toggle-track-check"
      />
      <div
        className="react-toggle-track-x"
      />
    </div>
    <div
      className="react-toggle-thumb"
    />
    <input
      checked={true}
      className="react-toggle-screenreader-only"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      type="checkbox"
    />
  </div>
</nav>
`;

exports[`NavBar renders correctly without logo 1`] = `
<nav
  className="sc-AxhCb dYJNSj navbar navbar-light"
>
  <div
    className="flex-fill"
  >
    <form
      className="mr-3"
    >
      <input
        aria-label="Search"
        autoFocus={true}
        className="sc-AxjAm gGRDat form-control"
        data-testid="search-bar"
        onChange={[Function]}
        placeholder="Search Courses "
        type="search"
        value=""
      />
    </form>
  </div>
  <div
    className="react-toggle react-toggle--checked"
    onClick={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
  >
    <div
      className="react-toggle-track"
    >
      <div
        className="react-toggle-track-check"
      />
      <div
        className="react-toggle-track-x"
      />
    </div>
    <div
      className="react-toggle-thumb"
    />
    <input
      checked={true}
      className="react-toggle-screenreader-only"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      type="checkbox"
    />
  </div>
</nav>
`;
